src/crimson/net/SocketMessenger.h

```diff
@@ -75,6 +79,10 @@ class SocketMessenger final : public Messenger {
     return shutdown_promise.get_future();
   }
 
+  seastar::future<> pre_shutdown() override {
```

Comment:
i think the purpose of `pre_shutdown()` is to close the "gate", so no dispatchers will be dispatched with more events, right?

Reply:
Oh, I think that's one operation that I missed in the pre_shutdown, I added pre_shutdown to unregister all dispatchers which seems to be problematic now.